### PR TITLE
Add validation of AWS account and region in environment creation

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Bugfix

### Detail
Added validation in environment creation. In this PR we add a check that raise an exception if an environment is created using the same AWS account and region as other environment.

### Relates
- #779 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Tested in AWS:
-[X] CICD pipeline completes
-[X] creating an environment using the same AWS Account/region throws exception
-[X] creating an environment using a differnet AWS Account/region succeeds

**Comment by dlpzx** [comment]
Something is off with the integration tests of this PR. The integration tests fail in the `core.environments.test_environment.test_archive_env` test, but the actual code does not correspond with the code executed in the test:

Real code:
image


Executed code:

image

File: backend/dataall/base/db/exceptions.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def __init__(self, param_name, param_value, constraint):
         self.param_name = param_name
         self.param_value = param_value
         self.message = f"""
-                    An error occurred (InvalidInput): f'{param_name} value {param_value} must be {constraint}'
+                    An error occurred (InvalidInput): '{param_name} value {param_value} must be {constraint}'
```
Comment (noah-paige) [inline backend/dataall/base/db/exceptions.py:57]: good fix 😀

**Review by noah-paige — APPROVED** [review]
approving

**Pull request merged**